docs/DeveloperGuide.md

```diff
@@ -153,6 +153,137 @@ The following sequence diagram shows how the undo operation works:
 
 _{Explain here how the data archiving feature will be implemented}_
 
+### Add Ingredient feature
+
+#### Implementation
+This feature allows users to add ingredients into the fridge.
+
+Substitutability is used in Command and Parser:
+* `AddIngredientCommand` extends `Command`
+* `AddIngredientCommandParser` implements `Parser<AddIngredientCommand>`
+
+Given below is an example usage scenario and how the mechanism behaves at each step.
+
+![AddIngredientSequence](images/AddIngredientSequence.png)
+
+Step 1:
+User inputs the add ingredient command to add ingredients into the fridge.
+
+Step 2:
+After successful parsing of user input, the `AddIngredientCommand#execute(Model model)` method is called.
+
+Step 3:
+The ingredients that the user has input will be saved into Wishful Shrinking's fridge of ingredients.
+
+Step 4:
+After the successful adding of the ingredient, a `CommandResult` object is instantiated and returned to `LogicManager`.
+
+#### Design Considerations
+##### Aspect 1: Concern while adding a new feature
+* Workflow must be consistent with other adding commands e.g. add recipe and eat recipe for consumption.
+
+##### Aspect 2: How do we successfully parse the ingredients the user has added with the optional ingredient quantity 
+* **Alternative 1 (current choice):** Add a quantity field in the Ingredient class as well as a IngredientParser class to parse each ingredient added
```

Comment:
actually the IngredientParser takes parses all the ingredients at once into an Ingredient arraylist. Maybe need to change the phrasing to something like parses the user input ingredients into Ingredient objects. For example if the command is addR n/salad c/20 instr/blah blah i/apple -200g, banana -2.5 cups then the parameter into IngredientParser will be "apple -200g, banana -2.5 cups" which will be parsed into ArrayList<Ingredient>{new Ingredient("apple", "200g"}, new Ingredient("banana", "2.5 cups")}